src/main/scala/com/microsoft/hyperspace/index/dataskipping/DataSkippingIndex.scala

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (2021) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.index.dataskipping
+
+import org.apache.spark.sql.{Column, DataFrame, SaveMode}
+import org.apache.spark.sql.functions.{input_file_name, min, spark_partition_id}
+
+import com.microsoft.hyperspace.index._
+import com.microsoft.hyperspace.index.dataskipping.sketch.Sketch
+import com.microsoft.hyperspace.index.dataskipping.util.ExpressionUtils
+import com.microsoft.hyperspace.util.HyperspaceConf
+
+/**
+ * DataSkippingIndex is an index that can accelerate queries by filtering out
+ * files in relations using sketches.
+ *
+ * @param sketches List of sketches for this index
+ * @param properties Properties for this index; see [[Index.properties]] for details.
+ */
+case class DataSkippingIndex(
+    sketches: Seq[Sketch],
+    override val properties: Map[String, String] = Map.empty)
+    extends Index {
```

Comment:
Option would be better than using special values like -1 or an empty string "". For containers such as Seq or Map, I think it depends. If the distinction between None and an empty container is needed, then you should go with Option, but if you don't care about the difference, then you can use an empty container instead of None.